### PR TITLE
Add receipt leaf components to Typescript definition

### LINKED ISSUE #3397: Add leaf components field to TypeScript `Receipt` type
Left-over from https://github.com/microsoft/CCF/pull/3292.

`js/ccf-app/src/global.ts` line 93

Also, I just noticed that `nodeId` may be incorrect, at least the C++ code says `node_id`.

### PR DESCRIPTION
**Pull request by eddyashton**
Resolves #3397.

### CODE REVIEW

File: src/js/historical.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ namespace ccf::js
     JS_SetPropertyStr(
       ctx,
       js_receipt,
-      "nodeId",
+      "node_id",
```
Comment (jumaffre) [inline src/js/historical.cpp:72]: I'm surprised no (end-to-end?) test broke because of this?

Comment (eddyashton) [reply src/js/historical.cpp:72]: Yeah we don't have great coverage of the JS/TS API, especially the recent Receipt changes. I'd like to expand that, but in a separate PRs.

Comment (eddyashton) [reply src/js/historical.cpp:72]: (Of course another way of saying that is "this PR is completely untested, but _looks ok_"...)

Comment (jumaffre) [reply src/js/historical.cpp:72]: It may be worth tracking this in a new (or existing) GH issue?

Comment (eddyashton) [reply src/js/historical.cpp:72]: #3503

Comment (achamayou) [reply src/js/historical.cpp:72]: I don't think the qjs coverage of receipts is any less than for C++, it's only v8 that's missing some bits. The node_id was once used before the PEM was in the receipts, to look up files, but isn't anymore.

It doesn't really make a lot of sense to have it _and_ the PEM, the thought was that we'd have full state historical queries, and make signatures more compact and do the lookup on receipt creation (or even leave it to the user), but it's not clear that this is the right tradeoff. We have a discussion scheduled about receipts, I think we'll want to cover that then.

**Pull request merged**